services/payments/tests/unit/test_services_auto_recharge_listener.py

```diff
@@ -78,3 +116,282 @@ async def test_process_event_functions(
     ):
         with attempt:
             mocked_message_parser.assert_called_once()
+
+
+@pytest.fixture()
+def populate_test_db(
+    postgres_db: sa.engine.Engine, faker: Faker, wallet_id: int
+) -> Iterator[None]:
+    with postgres_db.connect() as con:
+        _primary_payment_method_id = faker.uuid4()
+        _completed_at = datetime.now(tz=timezone.utc) + timedelta(minutes=1)
+
+        con.execute(
+            payments_methods.insert().values(
+                **random_payment_method(
+                    payment_method_id=_primary_payment_method_id,
+                    wallet_id=wallet_id,
+                    state="SUCCESS",
+                    completed_at=_completed_at,
+                )
+            )
+        )
+        con.execute(
+            payments_autorecharge.insert().values(
+                **random_payment_autorecharge(
+                    primary_payment_method_id=_primary_payment_method_id,
+                    wallet_id=wallet_id,
+                )
+            )
+        )
+
+        yield
+
+        con.execute(payments_methods.delete())
+        con.execute(payments_autorecharge.delete())
+
+
+@pytest.fixture()
+def wallet_id(faker: Faker):
+    return faker.pyint()
+
+
+@pytest.fixture
+async def mocked_pay_with_payment_method(mocker: MockerFixture) -> mock.AsyncMock:
+    return mocker.patch(
+        "simcore_service_payments.services.payments.PaymentsGatewayApi.pay_with_payment_method",
+        return_value=AckPaymentWithPaymentMethod.construct(
+            **AckPaymentWithPaymentMethod.Config.schema_extra["example"]
+        ),
+    )
+
+
+@pytest.fixture
+async def mock_rpc_server(
```

Comment:
consider separating in 2 fixtures. this one contains the server but you are returning the client

Reply:
@sanderegg guess who influenced this one 🙃

Reply:
Done :+1: and yes kudos to @GitHK :)